plugins/vpc-shared-eni/config/netconfig.go

```diff
@@ -30,33 +30,35 @@ import (
 // NetConfig defines the network configuration for the vpc-shared-eni plugin.
 type NetConfig struct {
 	cniTypes.NetConf
-	ENIName          string
-	ENIMACAddress    net.HardwareAddr
-	ENIIPAddress     *net.IPNet
-	VPCCIDRs         []net.IPNet
-	BridgeType       string
-	BridgeNetNSPath  string
-	IPAddress        *net.IPNet
-	GatewayIPAddress net.IP
-	InterfaceType    string
-	TapUserID        int
-	Kubernetes       KubernetesConfig
+	ENIName            string
+	ENIMACAddress      net.HardwareAddr
+	ENIIPAddresses     []net.IPNet
+	VPCCIDRs           []net.IPNet
+	BridgeType         string
+	BridgeNetNSPath    string
+	IPAddresses        []net.IPNet
+	GatewayIPAddress   net.IP
+	GatewayIPv6Address *net.IPNet
+	InterfaceType      string
+	TapUserID          int
+	Kubernetes         KubernetesConfig
 }
 
 // netConfigJSON defines the network configuration JSON file format for the vpc-shared-eni plugin.
 type netConfigJSON struct {
 	cniTypes.NetConf
-	ENIName          string   `json:"eniName"`
-	ENIMACAddress    string   `json:"eniMACAddress"`
-	ENIIPAddress     string   `json:"eniIPAddress"`
-	VPCCIDRs         []string `json:"vpcCIDRs"`
-	BridgeType       string   `json:"bridgeType"`
-	BridgeNetNSPath  string   `json:"bridgeNetNSPath"`
-	IPAddress        string   `json:"ipAddress"`
-	GatewayIPAddress string   `json:"gatewayIPAddress"`
-	InterfaceType    string   `json:"interfaceType"`
-	TapUserID        string   `json:"tapUserID"`
-	ServiceCIDR      string   `json:"serviceCIDR"`
+	ENIName            string   `json:"eniName"`
+	ENIMACAddress      string   `json:"eniMACAddress"`
+	ENIIPAddresses     []string `json:"eniIPAddresses"`
+	VPCCIDRs           []string `json:"vpcCIDRs"`
+	BridgeType         string   `json:"bridgeType"`
+	BridgeNetNSPath    string   `json:"bridgeNetNSPath"`
+	IPAddresses        []string `json:"ipAddresses"`
+	GatewayIPAddress   string   `json:"gatewayIPAddress"`
+	GatewayIPv6Address string   `json:"gatewayIPv6Address"`
```

Comment:
How much do we care about backwards compatibility here?  One alternative is to restructure the above 'addresses[]' arrays into a single array of multiple address/eni-address/gateway tuples.  ie:

```golang
type NetConfAddress struct {
        ENIIPAddress *net.IPNet
        IPAddress    *net.IPNet
        Gateway      net.IP
}

type NetConfig struct {
        ...
        Addresses    []*NetConfAddress
}
```

In particular, this allows code to loop over each NetConfAddress without having to know anything about address family, or other matching criteria.

Reply:
We have parked the refactor of this code for now, also the struct might have same gateway ip for different IPAddresses